CMakeLists.txt

```diff
@@ -292,6 +292,10 @@ if (NOT USE_SIMD STREQUAL "")
             else ()
                 add_definitions ("-m${feature}")
             endif ()
+            if (feature STREQUAL "fma" AND CMAKE_COMPILER_IS_CLANG)
+                # Clang needs an explicit flag to make fma ops
+                add_definitions ("-ffp-contract=fast")
```

Comment:
Since these functions are all in public headers, whatever decision we make here could be propagated to all users (or they would get worse performance).

Thats why the opt-in behavior seems safer to me. It lets this code take advantage of FMA when enabled, but also lets client programs build without `fp-contract=fast`.

Do you have any good pointers that document the difference between `fp-contract=fast` and `fp-contract=on` ?

I would also love to re-measure the performance/accuracy of our fast math approximations with FMA enabled.